sympy/combinatorics/free_group.py

```diff
@@ -806,6 +831,58 @@ def subword(self, from_i, to_j):
             array_form = letter_form_to_array_form(letter_form, group)
             return group.dtype(array_form)
 
+    def is_independent(self, word):
```

Comment:
It seems that the relators are scanned and generators that appear only once are marked redundant. Those relators that contain such generators are skipped over. After all relators have been scanned redundant generators are replaced in the relators by their representations in terms of other generators and deleted from the list. Then the process can be repeated. (Relators containing generators marked redundant cannot be used to decide if a generator is redundant. Even if it appears only once, it may reappear when another generator is replaced.)